Describe how to use Localstack to improve the development experience with AWS (#6875)
Particular/docs.particular.net: nservicebus/aws/local-development.md

---
title: AWS local development using LocalStack
summary: How to use LocalStack for your development needs when using Amazon SQS Transport and DynamoDB Persistence
reviewed: 2024-10-29
---

[LocalStack](https://www.localstack.cloud/) is a tool to develop and test your AWS applications locally, reducing development time and increasing productivity.

> [!NOTE]
> Refer to the [official LocalStack documentation](https://docs.localstack.cloud/getting-started/installation/) to learn how to install and run it locally.

To configure an NServiceBus endpoint to connect to LocalStack instead of AWS, the AWS endpoint URL must be set to the address of the LocalStack instance. The simplest way is by defining the `AWS_ENDPOINT_URL` environment variable and setting its value to the LocalStack URL:

```
AWS_ENDPOINT_URL=http://localhost.localstack.cloud:4566
```

Alternatively, configure the AWS SDK `ServiceURL` configuration property, like in the following example programmatically:

```csharp
var amazonSqsConfig = new AmazonSQSConfig();
amazonSqsConfig.ServiceURL = "http://localhost.localstack.cloud:4566";
var amazonSqsClient = new AmazonSQSClient(amazonSqsConfig);
```

> 

Reviewer: ```suggestion

The above example refers to Amazon SQS. The same needs to be done for Amazon SNS and Amazon DynamoDB when using these services.

```

---
title: AWS local development using LocalStack
summary: How to use LocalStack for your development needs when using Amazon SQS Transport and DynamoDB Persistence
reviewed: 2024-10-29
---

[LocalStack](https://www.localstack.cloud/) is a tool to develop and test your AWS applications locally, reducing development time and increasing productivity.

> [!NOTE]
> Refer to the [official LocalStack documentation](https://docs.localstack.cloud/getting-started/installation/) to learn how to install and run it locally.

To configure an NServiceBus endpoint to connect to LocalStack instead of AWS, the AWS endpoint URL must be set to the address of the LocalStack instance. The simplest way is by defining the `AWS_ENDPOINT_URL` environment variable and setting its value to the LocalStack URL:

```
AWS_ENDPOINT_URL=http://localhost.localstack.cloud:4566
```

Alternatively, configure the AWS SDK `ServiceURL` configuration property, like in the following example programmatically:

```csharp
var amazonSqsConfig = new AmazonSQSConfig();
amazonSqsConfig.ServiceURL = "http://localhost.localstack.cloud:4566";
var amazonSqsClient = new AmazonSQSClient(amazonSqsConfig);
```

> The above example refers to Amazon SQS. The same needs to be done for Amazon SNS and Amazon DynamoDB when using these services.

Reviewer: ```suggestion
The above example refers to Amazon SQS. When using these services, the same must be done for Amazon SNS and Amazon DynamoDB. More details are in the [Amazon SQS configuration options documentation](/transports/sqs/configuration-options.md).
```

---
title: AWS local development using LocalStack
summary: How to use LocalStack for your development needs when using Amazon SQS Transport and DynamoDB Persistence
reviewed: 2024-10-29
---

> [LocalStack](https://www.localstack.cloud/) is a tool to develop and test your AWS applications locally reducing development time and increasing product velocity.

Reviewer 2: ```suggestion
[LocalStack](https://www.localstack.cloud/) is a tool to develop and test your AWS applications locally, reducing development time and increasing product velocity.
```

> [!NOTE]
> Refer to the [official LocalStack documentation](https://docs.localstack.cloud/getting-started/installation/) to learn how to install and run it locally.

> To configure an NServiceBus endpoint to connect to LocalStack instead of connecting to AWS, the AWS endpoint URL must to be set to the address of the LocalStack instance.
> 
> There are several ways you can achieve that.

Reviewer 2: ```suggestion
To configure an NServiceBus endpoint to connect to LocalStack instead of connecting to AWS, the AWS endpoint URL must to be set to the address of the LocalStack instance. There are several ways you can achieve that.
```

The simplest way is by defining the following environment variable:

```
AWS_ENDPOINT_URL=http://localhost.localstack.cloud:4566
```

The alternative is to programmatically configure the AWS SDK like in the following example:

```csharp
var amazonSqsConfig = new AmazonSQSConfig();
amazonSqsConfig.ServiceURL = "http://localhost.localstack.cloud:4566";
var amazonSqsClient = new AmazonSQSClient(amazonSqsConfig);
```

> [!NOTE]
> Remember that, even if you are not connecting to AWS cloud services, it is still required to specify access credentials and region.
> From the LocalStack perspective they could be fake values, like the following:
>

> > ```

Reviewer 2: Don't know if it matters much but…

```suggestion
> ```bash
```
